converse.js

```diff
@@ -593,8 +596,10 @@
         this.reconnect = function (condition) {
             converse.log('Attempting to reconnect in 5 seconds');
             converse.giveFeedback(__('Attempting to reconnect in 5 seconds'), 'error');
-            setTimeout(function () {
+            clearTimeout(converse.reconnectTimeout);
+            converse.reconnectTimeout = setTimeout(function () {
                 if (converse.authentication !== "prebind") {
+                    this.connection.pass = this.password;
                     this.connection.connect(
```

Comment:
This looks wrong. There's no guarantee that `this.password` has a value, it's an optional config setting.